x-pack/elastic-agent/pkg/agent/application/pipeline/emitter/controller.go

```diff
@@ -163,7 +167,14 @@ func (e *Controller) update() error {
 	}
 
 	for _, r := range e.reloadables {
-		if err := r.Reload(cfg); err != nil {
+		// ast is filtered with capabilities etc.
+		if astR, ok := r.(astReloader); ok {
+			err = astR.ReloadAST(ast)
+		} else {
+			err = r.Reload(cfg)
+		}
```

Comment:
I don't think this is working as we want it to. Being that each program in the above `program.Programs` creates its own AST and then applied the rules this will not be the whole story.